functions/go/set-labels/kustomizerhelper.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"sigs.k8s.io/kustomize/api/konfig"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func ProcessLocalConfigResources(resources []*kyaml.RNode, fltr kio.Filter) ([]*kyaml.RNode, error) {
```

Comment:
Please add a godoc to explain what this method is doing.

Reply:
Why are we doing this in a function as opposed to orchestrator ? This is a common problem for all the functions correct ?

Reply:
This is PR is trying to fix a bug in functions that use `ResMap` which include set-annotations, set-labels, set-namespace and ensure-name-substring.
This bug affects `render`, for example https://github.com/GoogleContainerTools/kpt/tree/87191859fdef443395741d85e134e5aa1cd0f877/package-examples/pipeline-validate.

Input: 4 resources (2 normal resources + 2 local-config resources for gatekeeper)
==(set-label fn)==> 2 resources (2 normal resources)   # we expect the 2 local-config resources to be preserved
==(enforce-gatekeeper)==> 2 resources (2 normal resources) 

We can't really fix this bug in the orchestrator.